ds3-cli-integration/src/test/java/com/spectralogic/ds3cli/integration/test/helpers/ABMTestHelper.java

```diff
@@ -0,0 +1,456 @@
+import com.spectralogic.ds3client.commands.spectrads3.*;
+import com.spectralogic.ds3client.models.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.security.SignatureException;
+import java.util.UUID;
+
+import static org.apache.http.util.TextUtils.isEmpty;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+/**
+ * This class provides utilities for testing the ABM commands
+ */
+public final class ABMTestHelper {
+
+    final private static Logger LOG = LoggerFactory.getLogger(ABMTestHelper.class);
+
+    /**
+     * Creates a data policy with the specified name and versioning level, if a
+     * policy with the same name does not currently exist. If a policy already
+     * exists with the specified name, an error is thrown.
+     */
+    public static PutDataPolicySpectraS3Response createDataPolicyWithVersioning(
+            final String dataPolicyName,
+            final VersioningLevel versioningLevel,
+            final Ds3Client client) throws IOException, SignatureException {
+        return createDataPolicyWithVersioningAndCrcRequired(
+                dataPolicyName,
+                versioningLevel,
+                null,
+                client);
+    }
+
+    /**
+     * Creates a data policy with the specified name and versioning level and checksum, if a
+     * policy with the same name does not currently exist. If a policy already
+     * exists with the specified name, an error is thrown.
+     */
+    public static PutDataPolicySpectraS3Response createDataPolicyWithVersioningAndCrcRequired(
+            final String dataPolicyName,
+            final VersioningLevel versioningLevel,
+            final ChecksumType.Type checksumType,
+            final Ds3Client client) throws IOException, SignatureException {
+        //Check if data policy already exists
+        try {
+            client.getDataPolicySpectraS3(new GetDataPolicySpectraS3Request(dataPolicyName));
+            fail("Data policy already exists, terminating to prevent conflict: " + dataPolicyName);
+        } catch (final IOException e) {
+            //Pass: expected data policy to not exist
+        }
+
+        if (checksumType == null) {
+            //Create the data policy with versioning
+            return client.putDataPolicySpectraS3(new PutDataPolicySpectraS3Request(dataPolicyName)
+                    .withVersioning(versioningLevel));
+                    //.withAlwaysForcePutJobCreation(true));
+        }
+        //Create the data policy with versioning and checksum
+        return client.putDataPolicySpectraS3(new PutDataPolicySpectraS3Request(dataPolicyName)
+                .withVersioning(versioningLevel)
+                .withEndToEndCrcRequired(true)
+                .withChecksumType(checksumType));
+                //.withAlwaysForcePutJobCreation(true));
```

Comment:
I think its only available with 3.2+

We are shooting for 3.0.x compatibility at this point, right?  If so I'll go ahead and remove - just working through some issues to get all of the tests passing on a freshly PXE'd BP